src/main/java/com/facebook/openwifirrm/ucentral/UCentralClient.java

```diff
@@ -339,37 +441,52 @@ private HttpResponse<String> httpGet(
 		}
 	}
 
-	/** Send a POST request with a JSON body. */
+	/** Send a POST request with a JSON body and query params. */
 	private HttpResponse<String> httpPost(
 		String endpoint,
 		String service,
-		Object body
+		Object body,
+		Map<String, Object> query
 	) {
 		return httpPost(
 			endpoint,
 			service,
 			body,
+			query,
 			socketParams.connectTimeoutMs,
 			socketParams.socketTimeoutMs
 		);
 	}
 
-	/** Send a POST request with a JSON body using given timeout values. */
+	/** Send a POST request with a JSON body and query params using given timeout values. */
 	private HttpResponse<String> httpPost(
 		String endpoint,
 		String service,
 		Object body,
+		Map<String, Object> query,
 		int connectTimeoutMs,
 		int socketTimeoutMs
 	) {
 		String url = makeServiceUrl(endpoint, service);
-		HttpRequestWithBody req = Unirest.post(url)
-			.header("accept", "application/json")
-			.connectTimeout(connectTimeoutMs)
-			.socketTimeout(socketTimeoutMs);
+		HttpRequestWithBody req;
+		if (query == null || query.isEmpty()) {
+			req = Unirest.post(url)
+				.header("accept", "application/json")
+				.connectTimeout(connectTimeoutMs)
+				.socketTimeout(socketTimeoutMs);
+		} else {
+			req = Unirest.post(url)
+				.header("accept", "application/json")
+				.queryString(query)
+				.connectTimeout(connectTimeoutMs)
+				.socketTimeout(socketTimeoutMs);
+		}
```

Comment:
```suggestion
		HttpRequestWithBody req = Unirest.post(url)
				.header("accept", "application/json")
				.connectTimeout(connectTimeoutMs)
				.socketTimeout(socketTimeoutMs);
		if (query != null && !query.isEmpty()) {
			req.queryString(query);
		}
```

the above might be cleaner